sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/util/PassThroughWindowFn.java

```diff
@@ -30,24 +33,32 @@
 /**
  * A {@link WindowFn} that leaves all associations between elements and windows unchanged.
  *
- * <p>In order to implement all the abstract methods of {@link WindowFn}, this requires the
- * prior {@link WindowFn}, to which all auxiliary functionality is delegated.
+ * <p>This {@link WindowFn} is applied when elements must be passed through a {@link GroupByKey},
+ * but should maintain their existing {@link Window} assignments. Because windows may have been
+ * merged, the earlier {@link WindowFn} may not appropriately maintain the existing window
+ * assignments. This {@link WindowFn} allows existing window assigments to be maintained across
+ * a single group by key, at which point the earlier {@link WindowingStrategy} should be restored.
  *
  * <p>This {@link WindowFn} is an internal implementation detail of sdk-provided utilities, and
  * should not be used by {@link Pipeline} writers.
  */
 class PassThroughWindowFn<T> extends NonMergingWindowFn<T, BoundedWindow> {
 
-  /** The WindowFn prior to this. Used for its windowCoder, etc. */
-  private final WindowFn<?, BoundedWindow> priorWindowFn;
+  /**
+   * The coder of the type of windows of the input {@link PCollection}. This is not an arbitrary
+   * {@link BoundedWindow} {@link Coder}, but is safe to use for all windows assigned by this
+   * transform, as it should be the same coder used by the {@link WindowFn} that initially assigned
+   * these windows.
+   */
+  private Coder<BoundedWindow> coder;
 
-  public PassThroughWindowFn(WindowFn<?, ?> priorWindowFn) {
+  public PassThroughWindowFn(Coder<? extends BoundedWindow> coder) {
```

Comment:
May want to provide assignsToSingleWindow as well? And possibly some of the other properties.